Add more granularity to the ejection of process errors

Upon running workchains, we typically get things like this:

`exc_info = (<type 'exceptions.AttributeError'>, AttributeError("'int' object has no attribute 'status'",), None)`

`E AttributeError: 'int' object has no attribute 'status'`

Does not really matter what the error is (this is just an example). It would be great if it is possible to add more details to these ejections. Particularly when running tests on nested workchains it is very hard to know what failed (e.g. which workchain failed for instance). Or maybe there are ways to already do this that I am not seeing. If so I would really appreciate hints on how to proceed.

Indeed it is not. Here is a sample (of another error, which is easier to track down due to the explicit name):
```
>       results = run(workchain, **inputs)

aiida_vasp/workchains/tests/test_relax_wc.py:67: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../aiida_core/aiida/engine/launch.py:36: in run
    return runner.run(process, *args, **inputs)
../aiida_core/aiida/engine/runners.py:223: in run
    result, _ = self._run(process, *args, **inputs)
../aiida_core/aiida/engine/runners.py:211: in _run
    process.execute()
../../../env/aiida/lib/python2.7/site-packages/plumpy/processes.py:88: in func_wrapper
    return func(self, *args, **kwargs)
../../../env/aiida/lib/python2.7/site-packages/plumpy/processes.py:1063: in execute
    return self.future().result()
../../../env/aiida/lib/python2.7/site-packages/plumpy/futures.py:34: in result
    return super(Future, self).result(timeout)
../../../env/aiida/lib/python2.7/site-packages/tornado/concurrent.py:238: in result
    raise_exc_info(self._exc_info)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

exc_info = (<class 'aiida.common.exceptions.NotExistent'>, NotExistent('no neighbor with the label output_structure found',), None)

>   ???
E   NotExistent: no neighbor with the label output_structure found

<string>:3: NotExistent
```

Nothing before it. This is it.

--- a/docs/requirements_for_rtd.txt
+++ b/docs/requirements_for_rtd.txt
@@ -41,7 +41,7 @@ passlib==1.7.1
 pg8000<1.13.0
 pgtest==1.2.0
 pika==1.0.0
-plumpy==0.14.0
+plumpy==0.14.1
 psutil==5.5.1
 pyblake2==1.1.2; python_version<'3.6'
 pymatgen<=2018.12.12

--- a/environment.yml
+++ b/environment.yml
@@ -31,7 +31,7 @@ dependencies:
 - psycopg2==2.8
 - paramiko==2.4.2
 - ipython>=4.0,<6.0
-- plumpy==0.14.0
+- plumpy==0.14.1
 - kiwipy[rmq]==0.5.1
 - pika==1.0.0
 - circus==0.15.0

--- a/setup.json
+++ b/setup.json
@@ -43,7 +43,7 @@
     "psycopg2-binary==2.8",
     "paramiko==2.4.2",
     "ipython>=4.0,<6.0",
-    "plumpy==0.14.0",
+    "plumpy==0.14.1",
     "kiwipy[rmq]==0.5.1",
     "pika==1.0.0",
     "circus==0.15.0",

